torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -6943,10 +6943,65 @@ def test_ddp_multiple_nested_unused_params_err_ignore_params(self):
             # certain parameters.
             self._test_ddp_multiple_nested_unused_params_error(ignore_sparse=True)
 
-        @unittest.skipIf(
-            BACKEND != "nccl" and BACKEND != "gloo",
-            "Only Nccl & Gloo backend support DistributedDataParallel",
-        )
+        @unittest.skipIf(BACKEND != 'nccl' and BACKEND != 'gloo',
+                         "Only Nccl & Gloo backend support DistributedDataParallel")
+        @skip_if_lt_x_gpu(2)
+        def test_ddp_inference(self):
+            # tests that DDP module can be run on a single node with no_grad
+            # or eval setting and there is no hang.
+            rank = self.rank
+            torch.cuda.set_device(rank)
+            model = Net().cuda()
+            local_model = copy.deepcopy(model)
+            model = torch.nn.parallel.DistributedDataParallel(
+                model,
+                device_ids=[rank],
+            )
+            inp = torch.randn(10, 2, device=rank)
+            if self.rank == 0:
+                with torch.no_grad():
+                    for _ in range(6):
+                        ddp_out = model(inp)
```

Comment:
Nit: If you rename `model` as `ddp_model`, then you can just use one line here:
`self.assertEqual(ddp_model(inp), local_model(inp))`

Just more concise.